android-kit-base/src/main/java/com/mparticle/kits/CommerceEventUtils.java

```diff
@@ -56,7 +56,42 @@ public static List<MPEvent> expandProductAction(CommerceEvent event) {
         List<Product> products = event.getProducts();
         if (products != null) {
             for (int i = 0; i < products.size(); i++) {
-                MPEvent.Builder itemEvent = new MPEvent.Builder(String.format(ITEM_NAME, productAction), MParticle.EventType.Transaction);
+                MPEvent.Builder itemEvent;
+                switch (productAction) {
+                    case Product.ADD_TO_CART:
+                        itemEvent = new MPEvent.Builder(String.format(ITEM_NAME, productAction), MParticle.EventType.AddToCart);
```

Comment:
and this is exactly what customers do and is not correct. then we end up with weird data, broken filtering, integrations, etc.

MPEvent is custom event and cannot have AddToCart (or any of the others below) as a custom event type in our schema.